test/test_client_async.py

```diff
@@ -9,67 +11,55 @@
 
 logging.basicConfig(level=logging.WARNING)
 
+pytestmark = pytest.mark.asyncio
+
+
 ip = '127.0.0.1'
 tcpport = 1102
 db_number = 1
 rack = 1
 slot = 1
 
 
-class TestClient(unittest.TestCase):
-
-    @classmethod
-    def setUpClass(cls):
-        cls.process = Process(target=mainloop)
-        cls.process.start()
-        time.sleep(2)  # wait for server to start
-
-    @classmethod
-    def tearDownClass(cls):
-        kill(cls.server_pid, 1)
-
-    def setUp(self):
-        self.client = snap7.client.Client()
-        self.client.connect(ip, rack, slot, tcpport)
-
-    def tearDown(self):
-        self.client.disconnect()
-        self.client.destroy()
-
-    @unittest.skip("TODO: RuntimeWarning: coroutine 'TestClient.test_as_db_read' was never awaited")
-    async def test_as_db_read(self):
-        size = 40
-        start = 0
-        db = 1
-        data = bytearray(40)
-        self.client.db_write(db_number=db, start=start, data=data)
-        result = await self.client.as_db_read(db_number=db, start=start, size=size)
-        self.assertEqual(data, result)
-
-    @unittest.skip("TODO: RuntimeWarning: coroutine 'TestClient.test_as_db_write' was never awaited")
-    async def test_as_db_write(self):
-        size = 40
-        data = bytearray(size)
-        check = await self.client.as_db_write(db_number=1, start=0, data=data)
-        self.assertEqual(check, 0)
+@pytest.fixture(scope='module')
+def testserver():
+    process = Process(target=mainloop)
+    process.start()
+    time.sleep(2)  # wait for server to start
+    yield process
+    kill(process.pid, 1)
```

Comment:
doesn't this directly kill the server again

Reply:
This kills the server, if the test has finished...but I should have used the functional scope here.

Reply:
so this is only yielded at the end of the test?

Reply:
It's like the setUp/tearDown way in unittest.
"setUp" is everything before the yield and "tearDown" everything behind it.